components/textbox/textbox.tsx

```diff
@@ -66,257 +67,277 @@ type Props = {
 
-        this.updateSuggestions(prevProps);
-    }
-
-    checkMessageLength = (message: string) => {
-        if (this.props.handlePostError) {
-            if (message.length > this.props.characterLimit) {
-                const errorMessage = (
-                    <FormattedMessage
-                        id='create_post.error_message'
-                        defaultMessage='Your message is too long. Character count: {length}/{limit}'
-                        values={{
-                            length: message.length,
-                            limit: this.props.characterLimit,
-                        }}
-                    />);
-                this.props.handlePostError(errorMessage);
-            } else {
-                this.props.handlePostError(null);
-            }
-        }
-    }
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => onChange(e);
 
     // adding in the HTMLDivElement to support event handling in preview state
-    handleKeyDown = (e: KeyboardEvent<TextboxElement | HTMLDivElement>) => {
-        // since we do only handle the sending when in preview mode this is fine to be casted
-        this.props.onKeyDown?.(e as KeyboardEvent<TextboxElement>);
-    }
+    // since we do only handle the sending when in preview mode this is fine to be casted
+    const handleKeyDown = (e: KeyboardEvent<TextboxElement | HTMLDivElement>) => onKeyDown?.(e as KeyboardEvent<TextboxElement>);
 
-    handleSelect = (e: React.SyntheticEvent<TextboxElement>) => this.props.onSelect?.(e);
+    const handleSelect = (e: React.SyntheticEvent<TextboxElement>) => onSelect?.(e);
 
-    handleMouseUp = (e: MouseEvent<TextboxElement>) => this.props.onMouseUp?.(e);
+    const handleMouseUp = (e: MouseEvent<TextboxElement>) => onMouseUp?.(e);
 
-    handleKeyUp = (e: KeyboardEvent<TextboxElement>) => this.props.onKeyUp?.(e);
+    const handleKeyUp = (e: KeyboardEvent<TextboxElement>) => onKeyUp?.(e);
 
     // adding in the HTMLDivElement to support event handling in preview state
-    handleBlur = (e: FocusEvent<TextboxElement | HTMLDivElement>) => {
-        // since we do only handle the sending when in preview mode this is fine to be casted
-        this.props.onBlur?.(e as FocusEvent<TextboxElement>);
-    }
+    // since we do only handle the sending when in preview mode this is fine to be casted
+    const handleBlur = (e: FocusEvent<TextboxElement | HTMLDivElement>) => onBlur?.(e as FocusEvent<TextboxElement>);
 
-    handleHeightChange = (height: number, maxHeight: number) => {
-        this.props.onHeightChange?.(height, maxHeight);
-    }
+    const handleHeightChange = (height: number, maxHeight: number) => onHeightChange?.(height, maxHeight);
 
-    getInputBox = () => {
-        return this.message.current?.getTextbox();
-    }
+    const getInputBox = () => message.current?.getTextbox();
```

Comment:
Not strictly necessary, but it would be good if we could find a way to remove the need for the `getInputBox` entirely by forwarding a ref from this component's parent all the way to underlying input/textarea.

I'm not certain if we can actually do that though. I did that a while back with AutosizeTextarea and some other components around here (this line used to look like `this.message.current?.getTextbox().getInput().getInput()` lol), and there may have been a reason why I didn't get rid of this one.

Reply:
If that's not possible, minor nit then would be to move this into the `useImperativeHandle` below so that it's a bit more self-contained like
```js
useImperativeHandle(ref, () => {
    const getInputBox = ...;

    return {
        getInputBox,
        focus: () => {
             const textbox = getInputBox();
        },
    };
});
```